### PR TITLE
Rewrite pmem allocator test case and fix allocator bugs.

### PR DESCRIPTION
**Pull request by sherry-1001**
### What problem does this PR solve?
1. Rewrite pmem allocator test case: multi-thread allocate using a allocator instance.
2. Fix `stl set` compare function bug.
3. Optimize allocator fragmentation: reuse more pmem space.

Tests 

- Unit test
    dbtest_pmem_allocator run successfully (release/debug).
- Integration test

   bench performance：
  image

- No code

Side effects
    No.

### CODE REVIEW

File: tests/test_pmem_allocator.cpp [hunk 1]
```diff
@@ -51,56 +52,62 @@ TEST_F(EnginePMemAllocatorTest, TestBasicAlloc) {
...
+          ASSERT_EQ(space_entry.size != 0, true);
         }
-        ASSERT_EQ(kvpairs, alloc_cnt);
-      };
-
-      LaunchNThreads(num_thread, TestPmemAlloc);
-      delete pmem_alloc;
+        ASSERT_EQ(pmem_size / block_size, alloc_cnt);
+        delete pmem_alloc;
+        TearDown();
```
Comment (ZiyanShi) [inline tests/test_pmem_allocator.cpp:102]: TearDown() is the destructor of the test case. You should never call it manually.
Write another function to remove pmem file.

Comment (sherry-1001) [reply tests/test_pmem_allocator.cpp]: Done

**Review by ZiyanShi — APPROVED** [review]

File: engine/pmem_allocator/pmem_allocator.cpp [hunk 2]
```diff
@@ -115,6 +115,27 @@ PMEMAllocator* PMEMAllocator::NewPMEMAllocator(
     return nullptr;
   }
 
+  uint64_t segment_size = block_size * num_segment_blocks;
+
+  if (pmem_size < segment_size * max_access_threads) {
```
Comment (JiayuZzz) [inline engine/pmem_allocator/pmem_allocator.cpp:120]: please add this to configs check too

File: engine/pmem_allocator/pmem_allocator.cpp [hunk 3]
```diff
@@ -232,7 +244,7 @@ SpaceEntry PMEMAllocator::Allocate(uint64_t size) {
       if (palloc_thread_cache.free_entry.size >= aligned_size) {
         // Padding remaining space
         auto extra_space = palloc_thread_cache.free_entry.size - aligned_size;
-        if (extra_space >= kMinPaddingBlocks * block_size_) {
+        if (extra_space >= block_size_) {
```
Comment (JiayuZzz) [inline engine/pmem_allocator/pmem_allocator.cpp:247]: if(extra_space > 0);
and we can remove following "else" statement

File: engine/pmem_allocator/pmem_allocator.cpp (hunk 2 above)
Comment (sherry-1001) [reply engine/pmem_allocator/pmem_allocator.cpp:120]: has already exist.
image

**Review by ZiyanShi — APPROVED** [review]

**Pull request merged**